src/librustdoc/html/render/mod.rs

```diff
@@ -2101,112 +2140,121 @@ pub(crate) fn sidebar_render_assoc_items(
-            "Trait Implementations",
-            concrete_format.iter(),
-        );
-    }
-
-    if !synthetic_format.is_empty() {
-        print_sidebar_block(
-            out,
-            "synthetic-implementations",
-            "Auto Trait Implementations",
-            synthetic_format.iter(),
-        );
-    }
+#[derive(Template)]
+#[template(path = "sidebar/assoc_items.html")]
+struct AssocItems {
+    assoc_consts: Vec<SidebarLink>,
+    methods: Vec<SidebarLink>,
+    deref_methods: Vec<(Cow<'static, str>, String, Vec<SidebarLink>)>,
+    concrete: Vec<SideBarAssocItem>,
+    synthetic: Vec<SideBarAssocItem>,
+    blanket: Vec<SideBarAssocItem>,
+}
 
-    if !blanket_format.is_empty() {
-        print_sidebar_block(
-            out,
-            "blanket-implementations",
-            "Blanket Implementations",
-            blanket_format.iter(),
-        );
+impl AssocItems {
+    fn is_empty(&self) -> bool {
+        self.assoc_consts.is_empty()
+            && self.methods.is_empty()
+            && self.deref_methods.is_empty()
+            && self.concrete.is_empty()
+            && self.synthetic.is_empty()
+            && self.blanket.is_empty()
     }
 }
 
-fn sidebar_assoc_items(cx: &Context<'_>, out: &mut Buffer, it: &clean::Item) {
+fn sidebar_assoc_items(cx: &Context<'_>, it: &clean::Item) -> AssocItems {
     let did = it.item_id.expect_def_id();
     let cache = cx.cache();
 
+    let mut assoc_consts = Vec::new();
+    let mut methods = Vec::new();
     if let Some(v) = cache.impls.get(&did) {
         let mut used_links = FxHashSet::default();
         let mut id_map = IdMap::new();
 
         {
             let used_links_bor = &mut used_links;
-            let mut assoc_consts = v
-                .iter()
-                .filter(|i| i.inner_impl().trait_.is_none())
-                .flat_map(|i| get_associated_constants(i.inner_impl(), used_links_bor))
-                .collect::<Vec<_>>();
+            assoc_consts.extend(
+                v.iter()
+                    .filter(|i| i.inner_impl().trait_.is_none())
+                    .flat_map(|i| get_associated_constants(i.inner_impl(), used_links_bor)),
+            );
             if !assoc_consts.is_empty() {
                 // We want links' order to be reproducible so we don't use unstable sort.
                 assoc_consts.sort();
-
-                print_sidebar_block(
-                    out,
-                    "implementations",
-                    "Associated Constants",
-                    assoc_consts.iter(),
-                );
             }
-            let mut methods = v
-                .iter()
-                .filter(|i| i.inner_impl().trait_.is_none())
-                .flat_map(|i| get_methods(i.inner_impl(), false, used_links_bor, false, cx.tcx()))
-                .collect::<Vec<_>>();
+            methods.extend(
+                v.iter().filter(|i| i.inner_impl().trait_.is_none()).flat_map(|i| {
```

Comment:
The original code had each stage of this iterator pipeline on its own line. We should retain that, both because it's good style and because it minimizes diffs, making it easier to see what's changed.

Reply:
That change was `rustfmt`, I'll add a skip to this call